pkg/command/context.go

```diff
@@ -622,26 +641,65 @@ func globalTanzuLogin(c *configtypes.Context) error {
 
 	// format
 	fmt.Println()
+	printCSPOrgName(c, claims)
 	log.Success("successfully created a tanzu context")
 	return nil
 }
 
-func doCSPAuthAndUpdateContext(c *configtypes.Context, endpointType string) (claims *csp.Claims, err error) {
-	apiTokenValue, apiTokenExists := os.LookupEnv(config.EnvAPITokenKey)
+// printCSPOrgName prints the CSP Org name using the orgID from the claims.
+// It will only print the OrgName if the API is successful, if the API fails, it doesn't print anything
+func printCSPOrgName(c *configtypes.Context, claims *csp.Claims) {
+	issuer := csp.ProdIssuer
+	if staging {
+		issuer = csp.StgIssuer
+	}
+	orgName, err := csp.GetOrgNameFromOrgID(claims.OrgID, c.GlobalOpts.Auth.AccessToken, issuer)
+	if err != nil {
+		return
+	}
+	log.Infof("Successfully logged into '%s' organization \n", orgName)
+}
 
+func doCSPInteractiveLoginAndUpdateContext(c *configtypes.Context, endpointType string) (claims *csp.Claims, err error) {
 	issuer := csp.ProdIssuer
 	if staging {
 		issuer = csp.StgIssuer
 	}
-	if apiTokenExists {
-		log.Info("API token env var is set")
-	} else {
-		fmt.Fprintln(os.Stderr)
-		log.Info("The API key can be provided by setting the TANZU_API_TOKEN environment variable")
-		apiTokenValue, err = promptAPIToken(endpointType)
-		if err != nil {
-			return nil, err
-		}
+	// TODO(prkalle): Update the UX if necessary (may be introduce an extra command line option for orgID)
+	cspOrgIDValue, cspOrgIDExists := os.LookupEnv(constants.CSPLoginOrgID)
+	var options []csp.LoginOption
+	if cspOrgIDExists && cspOrgIDValue != "" {
+		options = append(options, csp.WithOrgID(cspOrgIDValue))
+	}
+	token, err := csp.TanzuLogin(issuer, options...)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get token from CSP for %s", endpointType)
```

Comment:
Thanks for pointining out. The `endpointType` parameter is not needed anymore( since we moved the API Token prompt to only apply for TMC login, earlier it was needed to prompt for API Token differently for TMC and Tanzu endpoint types). I will remove the `endpointType` parameter  and update the error message to `failed to get token from CSP`)